File: seminar_repos/sd_be/themeparks_part_2/supertest_1/db/seed.js
const db = require('./connection');
const format = require('pg-format');

const formatReviewsData = require('./utils');

function seed({ gameData, reviewData }) {
	return db
		.query('DROP TABLE IF EXISTS reviews;')
		.then(() => {
			return db.query('DROP TABLE IF EXISTS games;');
		})
		.then(() => {
			return createGames();
		})
		.then(() => {
			return createReviews();
		})
		.then(() => {
			return insertGames(gameData);
		})
		.then(({ rows: insertedGamesData }) => {
			const formattedReviews = formatReviewsData(reviewData, insertedGamesData);

			return insertReviews(formattedReviews);
		});
}

function createGames() {
	return db.query(
		`CREATE TABLE games (
			game_id SERIAL PRIMARY KEY,
			game_title VARCHAR NOT NULL,
			release_year INT,
			image_url VARCHAR, 
			console_name VARCHAR NOT NULL
			);`
	);
}

function createReviews() {
	return db.query(
		`CREATE TABLE reviews (
			review_id SERIAL PRIMARY KEY,
			game_id INT NOT NULL REFERENCES games(game_id),
			username VARCHAR NOT NULL,
			comment VARCHAR NOT NULL,
			rating INT
		);`
	);
}

function insertGames(gameData) {
	const gamesToInsert = gameData.map((game) => [
		game.game_title,
		game.release_year,
		game.image_url,
		game.console_name
	]);
	const queryStr = format(
		`INSERT INTO games
			(game_title, release_year, image_url, console_name)
			VALUES
			%L
		RETURNING *;`,
		gamesToInsert
	);
	return db.query(queryStr);
}

function insertReviews(reviewData) {
	const reviewsToInsert = reviewData.map((review) => [
		review.username,
		review.game_id,
		review.comment,
		review.rating
	]);
	const queryStr = format(
		`INSERT INTO reviews
			(username, game_id, comment, rating)
			VALUES
			%L
			RETURNING *;`,
		reviewsToInsert
	);
	return db.query(queryStr);
}

module.exports = seed;

File: seminar_repos/03_data_fun/05_rated_restaurants/01_fastapi/db/seed.py
from db.utils import create_games_lookup, format_reviews


def seed(conn, games, reviews):
    # Drop tables
    conn.run("DROP TABLE IF EXISTS reviews;")
    conn.run("DROP TABLE IF EXISTS games;")

    # Create tables
    conn.run("""
    CREATE TABLE games (
            game_id SERIAL PRIMARY KEY,
            game_title VARCHAR NOT NULL,
            release_year INT,
            console_name VARCHAR NOT NULL,
            image_url VARCHAR
            );
    """)

    conn.run("""
    CREATE TABLE reviews (
            review_id SERIAL PRIMARY KEY,
            game_id INT NOT NULL REFERENCES games(game_id),
            username VARCHAR NOT NULL,
            comment VARCHAR NOT NULL,
            rating INT
            );
    """)

    # Insert Games
    inserted_games = []

    for game in games:
        insert_query = f"""
        INSERT INTO games
        (game_title, release_year, console_name, image_url)
        VALUES
        (
                '{game['game_title']}',
                {game['release_year']},
                '{game['console_name']}',
                '{game['image_url']}'
        )
        RETURNING *;
        """
        inserted_row = conn.run(insert_query)
        inserted_games.append(inserted_row[0])

    games_id_lookup = create_games_lookup(inserted_games)
    formatted_reviews = format_reviews(reviews, games_id_lookup)

    # Insert Reviews
    for review in formatted_reviews:
        insert_query = f"""
        INSERT INTO reviews
        (game_id, username, comment, rating)
        VALUES
        (
                {review['game_id']},
                '{review['username']}',
                '{review['comment']}',
                {review['rating']}
        )
        """

        conn.run(insert_query)

    print("Data seeded...")
